### PR TITLE
Add support for minimal deployment of OCS

### PR DESCRIPTION
**Pull request by bipuladh**
Screenshot from 2020-07-30 23-02-23

/assign @cloudbehl

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts [hunk 1]
```diff
@@ -50,6 +51,23 @@ export const getOCSRequestData = (
...
+            memory: '8Gi',
+          },
+          requests: {
+            cpu: '1',
+            memory: '8Gi',
+          },
+        },
+      },
+    };
+  }
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts:69]: Have you checked this with OCS backend team?

File: frontend/packages/ceph-storage-plugin/src/utils/install.ts [hunk 2]
```diff
@@ -52,3 +57,14 @@ export const getAssociatedNodes = (pvs: K8sResourceKind[]): string[] => {
 
   return Array.from(nodes);
 };
+
+export const shouldDeployMinimally = (nodes: NodeKind[]) =>
+  nodes.reduce((acc, curr) => {
+    const cpus = humanizeCpuCores(getNodeCPUCapacity(curr)).value;
+    const memoryRaw = getNodeAllocatableMemory(curr);
+    const memory = humanizeBinaryBytes(convertToBaseValue(memoryRaw)).value;
+    if (memory < 61 && cpus < 16) {
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/utils/install.ts:66]: Make it 60 for safety & I have seen it.
```suggestion
    if (memory < 60 && cpus < 16) {
```

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx [hunk 3]
```diff
@@ -162,7 +170,14 @@ export const CreateOCS = withHandlePromise<CreateOCSProps & HandlePromiseProps>(
             <SelectNodesSection
               table={AttachedDevicesNodeTable}
               customData={{ filteredNodes, nodes, setNodes }}
-            />
+            >
+              {isMinimal && (
+                <div className="ceph-ocs-install__minimal-msg">
+                  Since the selected nodes do not satisfy the minimum requirements of 16 CPUs and 64
+                  GiB of RAM per node, a minimal cluster will be deployed.
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx:177]: As we don't know whether minimal will be also deployed or not.

I think we should communicate that we will try to deploy the minimal cluster with the specs provided.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts (hunk 1 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts:69]: yes. I and @oritwas talked about this. For now, we are moving forward with this. Tested too and is successfully getting deployed.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx [hunk 4]
```diff
@@ -162,7 +170,15 @@ export const CreateOCS = withHandlePromise<CreateOCSProps & HandlePromiseProps>(
             <SelectNodesSection
               table={AttachedDevicesNodeTable}
               customData={{ filteredNodes, nodes, setNodes }}
-            />
+            >
+              {isMinimal && (
+                <div className="ceph-ocs-install__minimal-msg">
+                  Since the selected nodes do not satisfy the recommended requirements of 16 CPUs
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx:176]: In BM/attached devices ocs install flow, we don't allow user to select nodes. `selected nodes` here doesn't seem correct.

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx:176]: The selected nodes are coming from Storage CLass right?

Comment (gnehapk) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx:176]: Yes, the nodes associated with the storage class are listed on the install page. My point is, UI doesn't allow selection of nodes on install page, so not sure whether naming it as selected nodes here is correct.

Comment (gnehapk) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx:176]: We can sync up with @yuvalgalanti on coming monday. For now lets go with this text, if required, will take this as a bug fix.

File: frontend/packages/ceph-storage-plugin/src/utils/install.ts [hunk 5]
```diff
@@ -52,3 +57,14 @@ export const getAssociatedNodes = (pvs: K8sResourceKind[]): string[] => {
 
   return Array.from(nodes);
 };
+
+export const shouldDeployMinimally = (nodes: NodeKind[]) =>
+  nodes.reduce((acc, curr) => {
+    const cpus = humanizeCpuCores(getNodeCPUCapacity(curr)).value;
+    const memoryRaw = getNodeAllocatableMemory(curr);
+    const memory = humanizeBinaryBytes(convertToBaseValue(memoryRaw)).value;
+    if (memory < 60 && cpus < 16) {
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/utils/install.ts:66]: Shouldn't this be OR condition?
```suggestion
    if (memory < 60 ||  cpus < 16) {
```

**Comment by spadgett** [comment]
> ERROR in /go/src/github.com/openshift/console/frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx
(135,80): Property 'then' does not exist on type 'void'.

Likely due to #6048

**Pull request merged**